ibis/backends/pandas/execution/arrays.py

```diff
@@ -44,28 +45,62 @@ def execute_array_index(op, data, index, **kwargs):
     )
 
 
-@execute_node.register(ops.ArrayIndex, list, int)
+@execute_node.register(ops.ArrayIndex, np.ndarray, int)
 def execute_array_index_scalar(op, data, index, **kwargs):
     try:
         return data[index]
     except IndexError:
         return None
 
 
-@execute_node.register(ops.ArrayConcat, pd.Series, (pd.Series, list))
-@execute_node.register(ops.ArrayConcat, list, pd.Series)
-@execute_node.register(ops.ArrayConcat, list, list)
-def execute_array_concat(op, left, right, **kwargs):
-    return left + right
+def _concat_iterables_to_series(
+    iter1: Iterable[Any], iter2: Iterable[Any],
+) -> pd.Series:
+    """Concatenate two iterables elementwise ("horizontally") to create a
+    Series. The two iterables are assumed to have the same length.
 
+    Used for ArrayConcat implementation.
+    """
+    assert len(iter1) == len(iter2)
+    result = pd.Series(map(lambda x, y: np.concatenate([x, y]), iter1, iter2))
+    return result
 
-@execute_node.register(ops.ArrayRepeat, pd.Series, pd.Series)
-@execute_node.register(ops.ArrayRepeat, int, (pd.Series, list))
-@execute_node.register(ops.ArrayRepeat, (pd.Series, list), int)
-def execute_array_repeat(op, left, right, **kwargs):
-    return left * right
+
+@execute_node.register(ops.ArrayConcat, pd.Series, pd.Series)
+def execute_array_concat_series(op, left, right, **kwargs):
+    return _concat_iterables_to_series(left, right)
+
+
+@execute_node.register(ops.ArrayConcat, pd.Series, np.ndarray)
+@execute_node.register(ops.ArrayConcat, np.ndarray, pd.Series)
```

Comment:
Can you comment here that the np.ndarray is for a single row and therefore we need to broadcast?